compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1147,6 +1147,7 @@ impl<'context> Elaborator<'context> {
         self.current_item = Some(DependencyId::Alias(alias_id));
         let typ = self.resolve_type(alias.type_alias_def.typ);
         self.interner.set_type_alias(alias_id, typ, generics);
+        self.generics.clear();
```

Comment:
I still prefer to always use recover_generics as a role of working with generics in the elaborator, but I see why this is simpler for type aliases.